src/devices/src/virtio/vsock/event_handler.rs

```diff
@@ -191,24 +192,32 @@ where
         let backend = self.backend.as_raw_fd();
         let activate_evt = self.activate_evt.as_raw_fd();
 
-        let mut raise_irq = false;
-
-        match source {
-            _ if source == rxq => raise_irq = self.handle_rxq_event(event),
-            _ if source == txq => raise_irq = self.handle_txq_event(event),
-            _ if source == evq => raise_irq = self.handle_evq_event(event),
-            _ if source == backend => {
-                raise_irq = self.notify_backend(event);
+        match self.device_state {
```

Comment:
Shouldn't this be using `is_activated` function instead?

Reply:
Sure. I've made the changes so all devices now use `is_activated()` in their event handler.